src/components/Header/Header.tsx

```diff
@@ -0,0 +1,55 @@
+import { useContext } from 'react';
+import { TodosContext } from '../../contextes/TodosContext';
+
+const reset = () => ({
+  id: +new Date(),
+  title: '',
+  completed: false,
+  editing: false,
+});
+
+export const Header = () => {
+  const {
+    todos,
+    setTodos,
+    newTodo,
+    setNewTodo,
+  } = useContext(TodosContext);
+
+  const changeValues = () => setTodos([
+    ...todos,
+    newTodo,
+  ]);
+
+  const handleOnSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (newTodo.title.trim()) {
+      changeValues();
+      setNewTodo(reset());
+    }
+  };
+
+  const hendlerOnChange
+  = (event: React.ChangeEvent<HTMLInputElement>) => setNewTodo({
+    ...newTodo,
+    title: event.target.value,
+  });
+
+  return (
+    <header className="header">
+      <h1>todos</h1>
+
+      <form onSubmit={event => handleOnSubmit(event)}>
```

Comment:
onSubmit, onClick, etc. always put an `event` in their callback, so if you put a function that receives that `event`, you can make this entry shorter
```suggestion
      <form onSubmit={handleOnSubmit}> // onSubmit={(event)} => onSubmit={handleOnSubmit}
```